bin/graceful_stop.sh

```diff
@@ -108,6 +108,10 @@ localhostname=`/bin/hostname -f`
 if [ "$localhostname" == "$hostname" ]; then
   local=true
 fi
+if [ "$localhostname" == "$hostname" ] || [ "$hostname" == "localhost" ]; then
+  local=true
+  hostname=$localhostname
```

Comment:
From my understanding, passing "localhost" to `regionserver.sh` is meant to be used in pseudo-distributed environments. See:
- https://issues.apache.org/jira/browse/HBASE-15924?focusedCommentId=16410243&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16410243
- https://issues.apache.org/jira/browse/HBASE-20130

And it would use fixed ports:
https://github.com/apache/hbase/blob/d74ae15fa7a110d3e2b35bf53d8823ba0b1bb33b/bin/regionservers.sh#L61-L64

Since I want to avoid that, let me reassign hostname to the fqdn here.

Reply:
> Do we have to repeat the 'if [ "$localhostname" == "$hostname" ] ...' and the setting of local = true

Ah no, that is my mistake. Sorry about that, we should remove it.